includes/objectcache/MemcachedClient.php

```diff
@@ -1351,6 +1368,67 @@ function _flush_read_buffer( $f ) {
 	// }}}
 	// }}}
 	// }}}
+
+	/**
+	 * Update memcached key in place to be standard-compliant. Overly long keys get shortened using sha1.
+	 *
+	 * @param $key string Memcached key to be checked and optionally fixed
+	 */
+	function _encode_key( &$key ) {
+		if ( is_array( $key ) ) {
+			$this->_encode_key( $key[1] );
+
+			return;
+		}
+
+		$key_length = strlen( $key );
+		if ( $key_length > self::MEMCACHED_KEY_MAX_LENGTH ) {
+			$suffix = '--HASH--' . sha1( $key );
+			$suffix_length = strlen( $suffix );
+			$key = substr( $key, 0, self::MEMCACHED_KEY_MAX_LENGTH - $suffix_length ) . $suffix;
```

Comment:
Will that meet the 250 character protocol constraint? Plus, it is 250 characters or 250 bytes?